RELEASE_CHECKLIST.md

```diff
@@ -0,0 +1,153 @@
+# Release Checklist
+
+This is a list of the things that need to happen during a release.
+
+## Build a Release
+
+### Prepare the Changelog (Full release only)
```

Comment:
I was a little confused at first What you meant by full release. Do you think there is a way to describe this before?

Reply:
i think i addressed this ok... let me know